tm2/pkg/bft/node/node.go

```diff
@@ -651,6 +669,11 @@ func (n *Node) OnStop() {
 	}
 }
 
+// FirstBlockReceived returns a channel that gets closed when the node receives its first block.
+func (n *Node) FirstBlockReceived() <-chan struct{} {
+	return n.firstBlockSignal
+}
+
```

Comment:
I've looked into this a bit more, and it turns out we can just use `isListening`, since it's set in`OnStart`. After `OnStart` finishes, the node should be fully set up and ready to go.

We can just move the `isListening` initialization down to the bottom of `OnStart`, and use the existing `Node.IsListening` to determine if the node is ready or not.

This way, we don't have to set up an event listener, and can be sure the node is setup if `IsListening` is true (it's reset when the node is closed down)

Reply:
Future Milos here:
It turns out this wasn't the case, the current code requires the first block to be committed in order for genesis state to be applied (consensus). I think this is out of the scope of this PR to fix